cpp/include/rapidsmpf/streaming/core/channel.hpp

```diff
@@ -58,11 +58,22 @@ class Channel {
      *
      * @return A coroutine that evaluates to the message, which will be empty if the
      * channel is shut down.
-     *
-     * @throws std::logic_error If the received message is empty.
      */
     coro::task<Message> receive();
 
+    /**
+     * @brief Asynchronously receive a message id from the channel.
+     *
+     * Suspends if the channel is empty. Once the message id is received, the message can
+     * be extracted using the `SpillableMessages::extract` method. This could be useful
+     * when a node wants to consume a message from a channel, but leave it in the
+     * spillable messages container for later extraction.
+     *
+     * @return A coroutine that evaluates to the message id. If the channel is shut down,
+     * the message id will be `SpillableMessages::InvalidMessageId`.
+     */
+    coro::task<SpillableMessages::MessageId> receive_message_id();
```

Comment:
I think the use of `MessageId` inside a `Channel` should remain an implementation detail. We should avoid exposing it.

Reply:
I still think this is useful to expose. Because channels no longer own the `Message`s. So, intuitively, we should be able to get the message or the placeholder of the message. Later would be useful when `Node`s decide to leave the messages in `SpillableMessages` and use it later on.

Reply:
The thing is, Channels **do** own their messages. They may store them in a `SpillableMessages` as an implementation detail, but the Channel still owns the messages until a `Node` receives them.

If we give up this isolation, a Channel loses the freedom to change its internal strategy for storing messages. For example, it might later decide to use multiple `SpillableMessages` instances if contention becomes a problem.

I am not saying we should never go in that direction, but that discussion should be a separate issue with its own clear motivation.